doc/tutorials/cmerge.md

```diff
@@ -0,0 +1,239 @@
+# How-To: kdb cmerge
+
+## Introduction
+
+The kdb tool allows users to access and perform functions on the Elektra key database from the command line. We added
+a new command to this very useful tool, the `cmerge` command. This command allows a user to perform a three-way merge
+of key sets from the `kdb` tool.
+
+The command to use this tool is:
+
+`kdb cmerge [OPTIONS] our their base result`
+
+`our`, `their` and `base` represent the three keys that are used in
+a three-way merge. Compared to the `diff3` tool, this would be
+the three files MYFILE (our), OLDFILE (base) and YOURFILE (their)
+that are the required for it. The result off the three-way merge
+will be stored in `result`.
+
+For configuration files in the context of package upgrades:
+
+- `their` is the current version of the maintainer's copy,
+- `base` is the previous version of the maintainer's copy.
+- `our` is the user's copy, derived from `base`
+
+## Options
+
+The options of `kdb cmerge`are:
+
+- `-f`, `--force`: overwrite existing keys in `result`
+- `-v`, `--verbose`: give additional information
+
+Strategies offer fine grained control over conflict handling. The option is:
+
+- `-s <name>`, `--strategy <name>`: which is used to specify a strategy to use in case of a conflict
+
+## Strategies
+
+The available strategies are:
+
+- `abort`: the merge will abort if any conflict happens and merge the 3 key sets together otherwise.
+- `our`: This option forces conflicting keys to be auto-resolved cleanly by favoring `our`. Changes from the other key sets that do not conflict with the `our` version are reflected to the merge result. This works like the recursive strategy with the `ours` option from git-merge.
+- `their`: This is the opposite of `our`. The merge will use the `their` version when a conflict happens.
+
+If no strategy is specified, the merge will default to the abort strategy.
+
+## Return value
+
+0 on success.
+
+1 if a conflict or any other error could not be resolved automatically.
```

Comment:
Make a different return value for "any other error".

Reply:
I now set the value 2 for if abort strategy is set and a conflict occured and 1 for "normal" errors like memory problems.